bindata/network/openshift-sdn/sdn-ovs.yaml

```diff
@@ -35,16 +35,31 @@ spec:
           #!/bin/bash
           set -euo pipefail
           export SYSTEMD_IGNORE_CHROOT=yes
+          # systemctl may not be used in a separate PID namespace to reach
+          # the systemd running in PID 1. Therefore we need to use the dbus API
+          systemctl_restart(){
+            gdbus call \
+              --system \
+              --dest org.freedesktop.systemd1 \
+              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
+              --method org.freedesktop.systemd1.Unit.Restart "replace"
+          }
+          svc_encode_name(){
+            # systemd encodes some characters, so far we only need to encode
+            # the character "-" but there may be more in the future.
+            echo "${1//-/_2d}"
+          }
+
           # Check to see if ovs is provided by the node:
-          if systemctl status ovsdb-server; then
+          if [[ -f '/etc/systemd/system/multi-user.target.wants/openvswitch.service' ]]; then
```

Comment:
I think Casey had a comment about this

https://github.com/openshift/cluster-network-operator/pull/672#issuecomment-647980159

> We could use is-enabled as a predicate, but that's not reliable for a few reasons - first of all, and this is the case on RHCOS, units without an [Install] section are defined as enabled (even though they're not). Second of all, we might not enable the unit directly in the future; we could make it a dependency of a meta-service. That would break this predicate again.

> I don't want to get in the business of checking for systemd configuration files on disk. They can live in a lot of places, so that's just setting ourselves up for trouble

Reply:
We have a big issue here, the problem of the `Failed to connect to bus: No data available` error happens both with gdbus and systemctl as we can see here:
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/25495/pull-ci-openshift-origin-master-e2e-gcp/1306219874485276672/artifacts/e2e-gcp/gather-extra/pods/openshift-sdn_ovs-5x4hr_openvswitch.log

We're two weeks aware from the code freeze and this is blocking several people. Debugging gdbus/systemctl will take time and the error is likely produced by underlying issues, and heavily dependant on timing. While I agree tihs solution isn't great as long as we can guarantee we're not enabling the ovs service in a different way during the 4.6 time frame we're pretty safe.

The code to run ovs outside 4.6 should be removed in 4.7 anyway so keeping this (which again is ugly) for 3 months doesn't look like such a bad decision...

Reply:
> The code to run ovs outside 4.6 should be removed in 4.7 anyway so keeping this (which again is ugly) for 3 months doesn't look like such a bad decision...

I'm not sure what you're suggesting? Are you saying we should drop this PR? That we should merge it as it is now?

Reply:
> I'm not sure what you're suggesting? Are you saying we should drop this PR?
No, what I'm trying to say is that because the fallback to start the ovs daemons in the container is only necessary for the 4.5 to 4.6 upgrade, eventually the vast majority of the start up script will be gone in 4.7.

> That we should merge it as it is now?
Yes, checking if the file is there has a bunch of problems but since it has such a limited scope I think it's viable.

Reply:
I'm fine with establishing a exact convention for detecting that host-OVS is enabled.   As long as MCO and CNO agree on the convention we should be okay, but this means we should put a warning in MCO to indicate that MCO should not change the `openvswitch.service` setting without updating CNO.

Reply:
Here maybe
https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/units/openvswitch.service.yaml#L2

Reply:
@danwinship @pliurh thoughts?

Reply:
@rbbratta you mean the location of the service file that we will check for here in the CNO yes?